src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/MarkDuplicatesSparkIntegrationTest.java

```diff
@@ -495,6 +496,33 @@ public void testAssertCorrectSortOrderMultipleBamsOverriding() {
         runCommandLine(args);
     }
 
+    @Test
+    public void testNullOpticalDuplicates() {
+        final File output = createTempFile("supplementaryReadUnmappedMate", "bam");
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        args.addOutput(output);
+        args.addInput(new File(TEST_DATA_DIR,"optical_dupes.bam"));
+        args.addFlag(MarkDuplicatesSpark.ALLOW_MULTIPLE_SORT_ORDERS_IN_INPUT_ARG);
+        args.add(OpticalDuplicatesArgumentCollection.READ_NAME_REGEX_LONG_NAME, "null");
```

Comment:
Is this really testing with an actual null pointer in the `READ_NAME_REGEX` field, or just the literal String "null"?

Can you add in a unit test showing that `OpticalDuplicateFinder` can be constructed from a null READ_NAME_REGEX, and that methods like `addLocationInformation()` can still function?

Reply:
Yes it is an actual `null` reference that gets passed. I believe the argument parser interprets "null" as `null` which is expected behavior in the picard MarkDuplicates documentation for disabling the OpticalDuplicates finder. In GATK4 we ported the same method but we accidentally encased it in a ternary that nulled out the Finder object (which should not have happened) and caused the exception.

Reply:
This test does actually assert that the optical dupes finder works as intended when "null" is provided. It is intended to count NOTHING as an optical duplicate and this test asserts that reads that normally would count as optical duplicates are treated as such. I'm somewhat remiss to add that test since the OpticalDuplicatesFinder object is a picard class that we call into in GATK.

Reply:
In picard `OpticalDuplicatesFinder` there is the following exit case for a null regex that is the intended target here: 
```
if (this.readNameRegex == null || length < 2 || length > maxDuplicateSetSize) {
            return opticalDuplicateFlags;
        }
````